doc/develop/getting_started/installation_linux.rst

```diff
@@ -77,7 +77,7 @@ need one.
       .. code-block:: console
 
          sudo apt-get install --no-install-recommends git cmake ninja-build gperf \
-           ccache dfu-util device-tree-compiler wget \
+           ccache dfu-util device-tree-compiler wget protobuf-compiler srecord \
```

Comment:
re: `protobuf-compiler`, the [nanopb sample documentation](https://github.com/zephyrproject-rtos/zephyr/blame/main/samples/modules/nanopb/README.rst#L19-L43), which supposedly is the only upstream sample that requires this, mentions this dependency.

Noting that, `protobuf-compiler` probably does not belong in the Getting Started Guide.

Reply:
Fair enough, do we have a policy on this?  I guess my thought is something along the lines of "Running twister with defaults settings on a host configured according to the intro docs should work".   Things that need external stuff should be non-default platforms, or maybe the dependencies should be made a module, etc...  But in general this seems like a bad user experience to me.

Reply:
If this is needed for Linux, then it should be needed for all the other platforms too, like Windows and macOS.

I agree with @stephanosio that requirements which apply to specific samples (or perhaps specific use cases) should be mentioned in documentation pages relevant to those boards/use cases/apps/etc.

Reply:
> Coming back to this: my concern was with the fact that these tests/platforms are twister defaults

The "ideal" solution to this problem would be to provide a way for the tests to specify the host tool dependencies (e.g. in `testcase.yaml`) and automatically skip tests if there is a missing dependency. We are already (sort of) doing that for the emulators; for example, if nSIM is not installed, twister only builds and automatically skips running.

Reply:
p.s. `srecord` looks reasonable to me; it is generic enough.

Reply:
What is the difference between the `protobuf` package in `scripts/requirements-extras.txt` and `protobuf-compiler`? If they are the same, I think it would be fine to mention that `pip` users may also wish to install the packages outlined in `scripts/requirements-extras.txt`.